cmd/thv-operator/controllers/mcpserver_controller.go

```diff
@@ -510,6 +510,30 @@ func (r *MCPServerReconciler) finalizeMCPServer(ctx context.Context, m *mcpv1alp
 		return err
 	}
 
+	// Step 2: Attempt to delete associated StatefulSet by name
+	sts := &appsv1.StatefulSet{}
+	err := r.Get(ctx, types.NamespacedName{Name: m.Name, Namespace: m.Namespace}, sts)
+	if err == nil {
+		// StatefulSet found, delete it
+		if err := r.Delete(ctx, sts); err != nil {
+			return fmt.Errorf("failed to delete StatefulSet %s: %w", m.Name, err)
```

Comment:
race condition: If the deletion happens between the `Get` and the `Delete`, this will error out. Do we want to check for `NotFound` error as part of the `Delete` error? This would address the race condition.